### ISSUE TITLE
New bgp_asn plan validation in aws_customer_gateway resources prevents use of 4-byte values

### ISSUE THREAD

**Issue by raxlarsbutler** [terraform-providers/terraform-provider-aws#14027]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.12.24

### Affected Resource(s)

* aws_customer_gateway

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behavior

The AWS API happily accepts 4-byte values for BGP ASNs (see https://docs.aws.amazon.com/vpn/latest/s2svpn/s2s-vpn-user-guide.pdf), so this should work as well with the Terraform AWS provider (and indeed, it does work prior to version 2.68).

### Actual Behavior

The recent changes (#13611) to the BGP ASN validation prevent the use of values outside of the interval [1, 65534]. This validation appears to assume that the BGP ASN is only two bytes. However, the current AWS API seems to support 4-byte ASNs, as defined in https://www.ietf.org/rfc/rfc4893.txt.

### Steps to Reproduce

1. Configure a Terraform project to use version 2.68 of the Terraform AWS provider.
2. Define an `aws_customer_gateway` resource with a `bgp_asn` > 65534. For example:

```terraform
resource "aws_customer_gateway" "cgw" {
  bgp_asn = 100000
  ...
}
```

3. Run `terraform plan`. This will yield an error: `Error: expected bgp_asn to be in the range (1 - 65534), got 100000`

The workaround I'm running with for now is to pin the provider version to `<2.68`. That works.

### References

* #13611
* https://github.com/terraform-providers/terraform-provider-aws/releases/tag/v2.68.0

**Comment by ewbankkit** [comment]
Although not clear from the [API documentation](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateCustomerGateway.html#API_CreateCustomerGateway_RequestParameters), the [user guide](https://docs.aws.amazon.com/vpn/latest/s2svpn/vpn-categories.html) is explicit about 4-byte ASNs:

> 4-byte ASN (in addition to 2-byte ASN)

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.69.0 of the Terraform AWS Provider, later today. Thanks to @ewbankkit for the fix. 👍
